### PR TITLE
Add Debugging Section to Testing Guides

### LINKED ISSUE #199: Add debugging section to LuckyFlow guides
@jwoertink Could you post here with some rough examples of what your team is using for different systems (Linux, Mac): https://gitter.im/luckyframework/Lobby?at=5de8468d08d0c961b7f06d31. We could likely use this in the debugging guide

I'd also love to link to this when LuckyFlow has issues connecting: https://gitter.im/luckyframework/Lobby?at=5dea763855066245987bdd39

### LINKED ISSUE #431: Document Lucky Flow's `take_screenshot` method
Perhaps we could add a little "Debugging" section to https://luckyframework.org/guides/testing, or just insert a new sub-item in https://luckyframework.org/guides/testing/html-and-interactivity.

It'd be nice to have something pop up in the guides when I searched "Screenshot". I ended up having to dig into the Lucky Flow source to find the method, but it's been really handy! Here's that definition: https://github.com/luckyframework/lucky_flow/blob/master/src/lucky_flow.cr#L48

### PR DESCRIPTION
**Pull request by stephendolan**
This PR aims primarily to close #431.
It also could technically close #199, though it doesn't incorporate any additional content from the comments of that issue.

This should serve as a framework we can easily expand as more folks encounter problems and issues are opened.

Here's what the page ended up looking like:
debugging_tests

### CODE REVIEW

File: src/actions/guides/testing/debugging_tests.cr [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        flow.sign_out
+        flow.sign_in "wrong-password"
+        flow.should_have_password_error
+        flow.sign_in "password"
+        flow.should_be_signed_in
+      end
+    end
+    ```
+
+    If we wanted to see what the state of the interface looks like after the `should_be_signed_in` step, all we need to do is insert a `take_screenshot` method call like so:
```
Comment (jwoertink) [inline src/actions/guides/testing/debugging_tests.cr]:
```suggestion
    If we wanted to see what the state of the interface looks like after the `flow.should_be_signed_in` step, all we need to do is insert a `flow.take_screenshot` method call like so:
```

Just for clarity

File: src/actions/guides/testing/debugging_tests.cr [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    ```crystal
+    require "../spec_helper"
+
+    describe "Authentication flow" do
+      it "works" do
+        flow = AuthenticationFlow.new("test@example.com")
+
+        flow.sign_up "password"
+        flow.should_be_signed_in
+        flow.take_screenshot
```
Comment (jwoertink) [inline src/actions/guides/testing/debugging_tests.cr:53]: Maybe a space before and after this to quicker highlight where that change is being made?

File: src/components/shared/header.cr [hunk 3]
```diff
@@ -55,8 +55,8 @@ class Shared::Header < BaseComponent
   private def nav_links
     nav_link("Guides", Guides::GettingStarted::Installing.path)
     nav_link("Blog", Blog::Index.path)
-    nav_link("Chat", "https://discord.gg/HeqJUcb")
-    nav_link("GitHub", "https://github.com/luckyframework/lucky")
+    nav_link("Chat", "https://discord.gg/HeqJUcb", target: "_blank")
```
Comment (jwoertink) [inline src/components/shared/header.cr]: wasn't this already done? 🤔 lol

**Review by jwoertink — COMMENTED** [review]
This will be super helpful. Especially with a template we can now start filling it in with some other stuff.

File: src/components/shared/header.cr (hunk 3 above)
Comment (stephendolan) [reply src/components/shared/header.cr]: Oops... no idea how that got into this branch. Apologies, I'll get that fixed up when I push your other suggestion.

File: src/actions/guides/testing/debugging_tests.cr (hunk 2 above)
Comment (stephendolan) [reply src/actions/guides/testing/debugging_tests.cr:53]: Added, and threw in a comment as well!

**Review by jwoertink — APPROVED** [review]

**Review by paulcsmith — COMMENTED** [review]
So one thng we may want to change here is use `open_screenshot` https://github.com/luckyframework/lucky_flow/blob/ce163cf7ce4050794c4aec2e4bdb840c620e4a70/src/lucky_flow.cr#L42

That way the user doesn't need to manually open it, it opens for you :D 

May want to still mention take_screenshot in case open_screeshot doesn't work on your system.

So maybe

* Change current docs to use `open_screenshot`
* Add section for `take_screenshot` if `open_screenshot` doesn't work on your system, or if you don't want it to open

Thoughts? @stephendolan

**Comment by stephendolan** [comment]
Created #451 !

**Pull request merged**